packages/grid/_modules_/grid/hooks/utils/useLogger.ts

```diff
@@ -54,27 +48,20 @@ type LoggerFactoryFn = (name: string) => Logger;
 let factory: LoggerFactoryFn | null;
 
 export function useLoggerFactory(
-  customLogger?: Logger | LoggerFactoryFn,
-  logLevel: string | boolean = process.env.NODE_ENV === 'production' ? 'error' : 'warn',
+  apiRef: any,
+  { logger, logLevel }: { logger?: Logger; logLevel?: string | false },
 ) {
   if (forceDebug) {
     factory = defaultFactory('debug');
     return;
   }
 
-  if (!customLogger) {
+  if (!logger) {
     factory = logLevel ? defaultFactory(logLevel.toString()) : null;
     return;
   }
 
-  if (isFunction(customLogger)) {
```

Comment:
Should this change be listed as a breaking change? Function-based loggers are no longer supported. Users have to provide an object.